### PR TITLE
Set up image shrinking and WebP conversion

### LINKED ISSUE #92: Optimize images
Compress the images, limit their resolution.

### PR DESCRIPTION
**Pull request by illright**
Closes #92

### CODE REVIEW

File: innopoints/core/image.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+from typing import Dict
+
+from PIL import Image
+
+from innopoints.core.helpers import abort
+
+
+def crop(image: Image.Image, dimensions: Dict[str, str]):
+    """Crop an image to the dimensions specified with keys 'x', 'y', 'width', 'height'."""
+    if 'x' not in dimensions:
```
Comment (aabounegm) [inline innopoints/core/image.py:12]: I don't understand, why the check for `x` explicitly, when you check for the rest of the keys in the try/catch?

File: innopoints/core/image.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        return image
+
+    try:
+        # pylint: disable=invalid-name
+        x = int(dimensions['x'])
+        y = int(dimensions['y'])
+        width = int(dimensions['width'])
+        height = int(dimensions['height'])
+    except KeyError:
+        abort(400, {'message': 'Not enough data to perform the crop.'})
```
Comment (aabounegm) [inline innopoints/core/image.py:22]: Should you really use `abort` from within a utility function? Wouldn't it be better to return `None` - for example - and check for it in the route function?

File: innopoints/core/image.py (hunk 1 above)
Comment (illright) [reply innopoints/core/image.py:12]: In some cases the cropping isn't desirable, so no keys are passed, it checks for this case

File: innopoints/core/image.py (hunk 2 above)
Comment (illright) [reply innopoints/core/image.py:22]: I thought about it, but it was ugly either way. Abort throws exceptions, so it's okay

File: innopoints/core/image.py (hunk 1 above)
Comment (aabounegm) [reply innopoints/core/image.py:12]: but this doesn't check for the absence of all keys, just `x`. What if everything else was passed except for it?

Comment (illright) [reply innopoints/core/image.py:12]: Then it's an invalid crop anyway

File: innopoints/core/image.py (hunk 2 above)
Comment (aabounegm) [reply innopoints/core/image.py:22]: I don't think that returning `None` or throwing a `ValueError` or something can be uglier than using `abort` outside a route function

File: innopoints/core/image.py (hunk 1 above)
Comment (aabounegm) [reply innopoints/core/image.py:12]: ok, so it shouldn't be silently ignored. It should be caught by the `except KeyError` that will rethrow an error

**Review by aabounegm — APPROVED** [review]

**Pull request merged**